extensions/vscode/src/views/homeView.ts

```diff
@@ -1140,12 +1240,22 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
     const activeDeployment = this.getActiveContentRecord();
     if (activeDeployment) {
       try {
-        const response = await api.files.getByConfiguration(
+        const apiRequest = api.files.getByConfiguration(
           activeDeployment.configurationName,
           {
             dir: activeDeployment.projectDir,
           },
         );
+        window.withProgress(
+          {
+            title: "ReFreshing Files",
+            location: { viewId: Views.HomeView },
+          },
+          async () => {
+            return apiRequest;
+          },
+        );
```

Comment:
I wonder if each of our views should have a helper to show progress like the below:

```
  private showProgress(promise: Promise<unknown>, title: string) {
    return window.withProgress(
      {
        title: title,
        location: { viewId: Views.HomeView },
      },
      async () => promise,
    );
  }
```

Could make it a bit less verbose each time.